src/main/java/uk/gov/hmcts/cmc/claimstore/services/OfferResponseDeadlineCalculator.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmcts.cmc.claimstore.services;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+@Component
+public class OfferResponseDeadlineCalculator {
+
+    private final WorkingDayIndicator workingDayIndicator;
+    private final int offerMadeTimeForResponseInDays;
+    private final int endOfBusinessDayHour;
+
+    public OfferResponseDeadlineCalculator(
+        WorkingDayIndicator workingDayIndicator,
+        @Value("${dateCalculations.offerMadeTimeForResponseInDays}") final int offerMadeTimeForResponseInDays,
+        @Value("${dateCalculations.endOfBusinessDayHour}") final int endOfBusinessDayHour) {
+
+        this.workingDayIndicator = workingDayIndicator;
+        this.offerMadeTimeForResponseInDays = offerMadeTimeForResponseInDays;
+        this.endOfBusinessDayHour = endOfBusinessDayHour;
+    }
+
+    public LocalDate calculateOfferResponseDeadline(LocalDateTime offerSentAt) {
```

Comment:
there's a mix of final parameters and non final in this class?

Reply:
all are `final` now